src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/client/IdamOpenIdClient.java

```diff
@@ -98,10 +102,13 @@ public String getBearerToken() {
 
     }
 
-
-    private String nextUserEmail() {
-        return String.format(testConfig.getGeneratedUserEmailPattern(),
-                RandomStringUtils.randomAlphanumeric(10));
+    public static String generateSidamPassword() {
+        String regex = "^(?=.{10,})(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9]).*$";
+        String password = RandomStringUtils.randomAlphanumeric(10);
+        if (!password.matches(regex)) {
+            password = generateSidamPassword();
+        }
+        return password;
```

Comment:
i think this code is not required here. just add the password in the key-vault and get it from there so that no need to add extra code . it's suggestion otherwise generate password once and assign to global variable so that you can reuse same for all the test cases instead of generating every time

Reply:
Used Generex library which will generate the password once and can be reused.